aws/lambdas/readMatchDetails/index.js

```diff
@@ -92,94 +85,77 @@ function getStats(matchChronology, homeTeam, awayTeam) {
         
         player.stats.fouls = fouls;
     });
-}
+};
 
-function getScores(element) {
+const getScores = (element) => {
     return $(element)
-            .find('td')
-            .last()
-            .text()
-            .trim()
-            .split(',')
-            .map(Number);
-}
-
-function getPlayers(homeTeam, awayTeam) {
-    const matchPage = $('.match_details_table').eq(0).find('tr')
-        .filter(function () {
-            return $(this).find('a').length > 0;
-        }).next();
-
-    homeTeam.id = getTeamId(matchPage.prev().find('a').eq(0));
-    awayTeam.id = getTeamId(matchPage.prev().find('a').eq(1));
-
-    let playerRow = matchPage.eq(0);
-
-    while (playerRow.length > 0) {
-        const homePlayerCell = playerRow.find('td').eq(0);
-
-        if (homePlayerCell.find('a').text() !== '') {
-            homeTeam.players.push({
-                name: getName(homePlayerCell.find('a')),
-                id: getQueryParams(homePlayerCell.find('a'))[0].split('=')[1],
-                stats: {
-                    fouls: 0,
-                    periods: []
-                }
-            });
-        }
-        
-        const awayPlayerCell = playerRow.find('td').eq(1);
-
-        if (awayPlayerCell.find('a').text() !== '') {
-            awayTeam.players.push({
-                name: getName(awayPlayerCell.find('a')),
-                id: getQueryParams(awayPlayerCell.find('a'))[0].split('=')[1],
-                stats: {
-                    fouls: 0,
-                    periods: []
-                }
-            });
-        }
-
-        playerRow = playerRow.next();
-    }
-
-    function getTeamId(element) {
-        return getQueryParams(element)[0].split('=')[1];
-    }
-
-    function getName(element) {
-        return element.text().split('(')[0].trim();
-    }
-}
+        .find('td')
+        .last()
+        .text()
+        .trim()
+        .split(',')
+        .map(Number);
+};
 
-const parseMatchPage = (html, matchId, eventId, matchDetailsLink) => {
-    $ = cheerio.load(html);
+const getPlayers = (homeTeam, awayTeam) => {
+    const getTeamId = (element) => getQueryParams(element)[0].split('=')[1];
+    const getName = (element) => element.text().split('(')[0].trim();
 
-    const matchChronology = $('.match_details_table').filter((index, element) => {
-        return $(element).find('h6').text() === 'Meccs kronológia';
-    });
+    const matchPageRows = $('.match_details_table').eq(0).find('tr:has(a)');
 
-    const matchTable = $('.match_details_table').filter((index, element) => {
-        return $(element).find('h6').text() === 'Meccslap';
+    matchPageRows.each((index, element) => {
+        if (index === 0) {
+            homeTeam.id = getTeamId($(element).find('a').eq(0));
+            awayTeam.id = getTeamId($(element).find('a').eq(1));
+        } else {
+            const homePlayerCell = $(element).find('td:nth-child(1):has(a:not(:empty))');
```

Comment:
I should really learn the css selectors... I cannot imagine how I would have tried to write this.
Elegant solution:)